protocol-definitions/DynamicConfig.yaml

```diff
@@ -1292,3 +1292,37 @@ methods:
           doc: |
             Properties of the data connection configuration.
     response: {}
+  - id: 18
+    name: addWanReplicationConfig
+    since: 2.7
+    doc: |
+      Adds a WAN replication configuration.
+    request:
+      retryable: false
+      partitionIdentifier: -1
+      params:
+        - name: name
+          type: String
+          nullable: false
+          since: 2.7
+          doc: |
+            WAN replication configuration name.
+        - name: consumerConfig
+          type: Data
```

Comment:
If you want backward compatibility guarantees, then try to avoid using Data as the type except the user provided fields which we have no control on, e.g. the user key, value for the data. Hence, I would suggest that you encode every field as you can into the message as a field. This will be more work to begin but it pays in the long term. Especially, also consider that this codec may be used by other non-Java clients as well. We can talk about this.